.github/workflows/integration_test.yaml

```diff
@@ -7,3 +7,9 @@ jobs:
   integration-tests:
     uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
     secrets: inherit
+    with:
+      pre-run-script: tests/integration/localstack-installation.sh
+      image-build-args: |
+        image_flavour=canonical
```

Comment:
In general I fully agree with your point of view.  
I think that the image_flavour can be handled at deploy time but I didn't manage to do it for local_mode.  

For now, this is out of scope in my opinion since I'm trying to keep it retrocompatible and I don't want to make to much changes before the conversion to sidecar. Let's have this discussion again after the conversion (in which I hope to improve the way the image is constructed).